File: app/src/main/java/com/datarockets/mnchkn/fragments/charts/di/ChartsComponent.java
package com.datarockets.mnchkn.fragments.charts.di;

import com.datarockets.mnchkn.di.ActivityScope;
import com.datarockets.mnchkn.di.AppComponent;
import com.datarockets.mnchkn.fragments.charts.ChartsFragment;

import dagger.Component;

@ActivityScope
@Component(dependencies = AppComponent.class, modules = ChartsModule.class)
public interface ChartsComponent {
    void inject(ChartsFragment chartsFragment);
}

File: app/src/main/kotlin/com/datarockets/mnchkn/fragments/charts/di/ChartsComponent.kt
package com.datarockets.mnchkn.fragments.charts.di

import com.datarockets.mnchkn.di.ActivityScope
import com.datarockets.mnchkn.di.AppComponent
import com.datarockets.mnchkn.fragments.charts.ChartsFragment

import dagger.Component

@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(ChartsModule::class))
interface ChartsComponent {
    fun inject(chartsFragment: ChartsFragment)
}
